### PR TITLE
api: Use OCI types for Digest and Platform

### PR DESCRIPTION
**Pull request by aaronlehmann**
The current use of the types from distribution brings in some
unfortunate dependencies, including other distribution packages and the
gorilla/mux and gorilla/context packages. Using the OCI types avoids
the extra dependencies for client users.

cc @nishanttotla @stevvooe @ijc25

### CODE REVIEW

**Comment by ijc25** [comment]
LGTM

File: api/server/router/distribution/distribution_routes.go
Comment (nishanttotla) [inline api/server/router/distribution/distribution_routes.go]: Do we not need to update the fourth field here?

**Comment by nishanttotla** [comment]
LGTM (I am not a maintainer).

This will impact https://github.com/moby/moby/pull/32388, so either that (or this) PR need to be updated depending on which gets merged first.

File: api/server/router/distribution/distribution_routes.go
Comment (stevvooe) [inline api/server/router/distribution/distribution_routes.go]: Don't use the features field at all. We are removing it.

Comment (aaronlehmann) [inline api/server/router/distribution/distribution_routes.go]: Updated

**Comment by stevvooe** [comment]
LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM (if green)

**Pull request merged**
